internal/converter/paths.go

```diff
@@ -14,44 +14,182 @@ import (
 )
 
 func addPathItemsFromFile(opts options.Options, fd protoreflect.FileDescriptor, paths *v3.Paths) error {
-	services := fd.Services()
-	for i := 0; i < services.Len(); i++ {
-		service := services.Get(i)
-		methods := service.Methods()
-		for j := 0; j < methods.Len(); j++ {
-			method := methods.Get(j)
-			pathItems := googleapi.MakePathItems(opts, method)
-			for pair := pathItems.First(); pair != nil; pair = pair.Next() {
-				path, item := pair.Key(), pair.Value()
-				if existing, ok := paths.PathItems.Get(pair.Key()); !ok {
-					paths.PathItems.Set(path, item)
-				} else {
-					if item.Get != nil {
-						existing.Get = item.Get
+		services := fd.Services()
+		for i := 0; i < services.Len(); i++ {
+			service := services.Get(i)
+			methods := service.Methods()
+			for j := 0; j < methods.Len(); j++ {
+				method := methods.Get(j)
+				pathItems := googleapi.MakePathItems(opts, method)
+				
+				// Helper function to update or set path items
+				updatePathItem := func(path string, newItem *v3.PathItem) {
```

Comment:
The name of this method is a bit misleading. This does the merging logic but also adds the path to the list. I'd recommend naming this `addPathItem`. If you change the signature to this you can pull it out and avoid using a closure:

```go
func addPathItem(paths *orderedmap.Map[string, *v3.PathItem], path string, item *v3.PathItem)
```